portfolio/src/main/java/com/google/sps/servlets/MapMarkerServlet.java

```diff
@@ -0,0 +1,98 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.gson.Gson;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+import java.util.List;
+
+/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+@WebServlet("/map-marker")
+public class MapMarkerServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    Query query = new Query("map-marker").addSort("timestamp", SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    ArrayList<ArrayList<Double>> pairList = new ArrayList<ArrayList<Double>>();
+    
+
+    for (Entity entity : results.asIterable()) {  
+        ArrayList<Double> latLngPair = new ArrayList<>();
+        Double latEntry = (Double) entity.getProperty("lat");
+        Double lngEntry = (Double) entity.getProperty("lng");
+        latLngPair.add(latEntry);
+        latLngPair.add(lngEntry);
+        pairList.add(latLngPair);
+    }
+
+    String json = convertJsonUsingGson(pairList);
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    
+    ArrayList<Double> pairLatLng = getCoords(request);
+    Long timestamp = System.currentTimeMillis();
+    Entity markerEntity = new Entity("map-marker");
+    markerEntity.setProperty("lat", pairLatLng.get(0));
+    markerEntity.setProperty("lng", pairLatLng.get(1));
+    markerEntity.setProperty("timestamp", timestamp);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(markerEntity);
+  }
+
+  private String convertJsonUsingGson(ArrayList<ArrayList<Double>> latLngPair) {
+    String json = new Gson().toJson(latLngPair);
+    return json;
+  }
+
+  private ArrayList<Double> getCoords(HttpServletRequest request) {
```

Comment:
Here I would say that it is better to return an object that has a lat and lng member instead of an array list because with the array list you need to remember that lat is index 0 and lng is index 1 whereas with the object the lat/lng members are self documentation and makes the code more readable.